# Data Analysis Solutions - Peter Kariuki

Import python libraries and data

In [1]:
import pandas as pd

In [2]:
fullfillment = pd.read_csv("fullfillment-data-example.csv")
inventory = pd.read_csv("inventory-data-example.csv")
order = pd.read_csv("order-data-example.csv")

In [3]:
fullfillment["source"] = "Fulfilment"
inventory["source"] = "Inventory"
order["source"] = "Order"


Preview the data provided

In [4]:
fullfillment.head(4)

order_date  order_id                         product_name  quantity  \
0  8/1/2019 12:56     10001          Supa Sanitary Pads 10Pieces       1.0   
1  8/1/2019 12:56     10002  Always Maxi Thick Extra Long 7 Pads       1.0   
2  8/1/2019 13:01     10003           Tembo Soap (2 pieces)*224g       2.0   
3  8/1/2019 13:01     10004          Supa Sanitary Pads 10Pieces       1.0   

   product_sales_total      source  
0                800.0  Fulfilment  
1               1000.0  Fulfilment  
2                600.0  Fulfilment  
3                800.0  Fulfilment

In [5]:
order.columns

Index(['order_date', 'order_id', 'product_name', 'quantity',
       'product_sales_total', 'source'],
      dtype='object')

In [6]:
order.columns

Index(['order_date', 'order_id', 'product_name', 'quantity',
       'product_sales_total', 'source'],
      dtype='object')

In [7]:
fullfillment.columns

Index(['order_date', 'order_id', 'product_name', 'quantity',
       'product_sales_total', 'source'],
      dtype='object')

In [8]:
inventory.columns


Index(['sales_date', 'order_id', 'product_name', 'quantity',
       'total_sales_amount', 'source'],
      dtype='object')

Rename the columns to be similar for merging

In [9]:
inventory.columns = ['order_date', 'order_id', 'product_name', 'quantity',
       'product_sales_total', 'source']
inventory.columns

Index(['order_date', 'order_id', 'product_name', 'quantity',
       'product_sales_total', 'source'],
      dtype='object')

In [10]:
order.head(4)


order_date  order_id                              product_name  \
0  8/1/2019 3:40     10001    VWash Plus Intimate Hygiene Wash 100ml   
1  8/1/2019 7:25     10002       O.B Original Super Tampons 8 Pieces   
2  8/1/2019 7:30     10003                   Best Ladies Maxi 10Pads   
3  8/1/2019 7:36     10004  Geisha Bath Soap Soothing Aloe Vera 225g   

   quantity  product_sales_total source  
0         1                 5000  Order  
1         1                 2100  Order  
2         1                  600  Order  
3         1                  700  Order

Check the dimention of the data files

In [11]:
fullfillment.shape
inventory.shape
order.shape

(100, 6)

## 1. Merge the data files

Merge fullfillment and inventory data files

In [12]:
fullfillment_inventory = fullfillment.append(inventory,sort=True)

Merge fullfillment and inventory and order data files

In [13]:
fullfillment_inventory_order = fullfillment_inventory.append(order,sort=True)
fullfillment_inventory_order.shape

(301, 6)

Export the data 

In [14]:
fullfillment_inventory_order.to_csv("fullfillment_inventory_order.csv")

In [15]:
fullfillment_inventory_order.head(3)

order_date  order_id                         product_name  \
0  8/1/2019 12:56     10001          Supa Sanitary Pads 10Pieces   
1  8/1/2019 12:56     10002  Always Maxi Thick Extra Long 7 Pads   
2  8/1/2019 13:01     10003           Tembo Soap (2 pieces)*224g   

   product_sales_total  quantity      source  
0                800.0       1.0  Fulfilment  
1               1000.0       1.0  Fulfilment  
2                600.0       2.0  Fulfilment

Drop NA's in the data frame since we have very few NA's in the data frame. Imputation is another option to deal with the problem.

In [16]:
fullfillment_inventory_order_na_dropped = fullfillment_inventory_order.dropna()
fullfillment_inventory_order_na_dropped.shape

(300, 6)

## 2. For this exercise use python to clean, merge and produce statistics on

### a) Total Number of Orders according to each source

In [17]:
fullfillment_inventory_order_na_dropped['source'].value_counts()

Order         100
Inventory     100
Fulfilment    100
Name: source, dtype: int64

### b) Total Revenue according to each source

In [18]:
fullfillment_inventory_order_na_dropped.groupby('source')['product_sales_total'].sum()

source
Fulfilment    251510.0
Inventory     240778.0
Order         544180.0
Name: product_sales_total, dtype: float64

### c) Products sold according to each source

In [19]:
fullfillment_inventory_order_na_dropped.groupby('source')['quantity'].sum()

source
Fulfilment    269.0
Inventory     353.0
Order         632.0
Name: quantity, dtype: float64

### d) Avg Basket Size according to each source

Assuming backet size is measured by the quantity.

In [20]:
fullfillment_inventory_order_na_dropped.groupby('source')['quantity'].mean()

source
Fulfilment    2.69
Inventory     3.53
Order         6.32
Name: quantity, dtype: float64

Assuming backet size is measured by the product sales.

In [21]:
fullfillment_inventory_order_na_dropped.groupby('source')['product_sales_total'].mean()

source
Fulfilment    2515.10
Inventory     2407.78
Order         5441.80
Name: product_sales_total, dtype: float64

## 3 . Summarize the differences are across each data source ie # of orders, revenue, products sold.

#### Number of orders - differences 

In [22]:
fullfillment_inventory_order_na_dropped['source'].value_counts()

Order         100
Inventory     100
Fulfilment    100
Name: source, dtype: int64

The three sources had an equal number of orders.

#### Revenue differences 

In [23]:
fullfillment_inventory_order_na_dropped.groupby('source')['product_sales_total'].sum()

source
Fulfilment    251510.0
Inventory     240778.0
Order         544180.0
Name: product_sales_total, dtype: float64

The difference between Order and Fulfilment  is 292670 (544180 - 251510)

The difference between Inventory and Fulfilment  is 303402 (544180 - 240778)

#### Products sold

In [24]:
fullfillment_inventory_order_na_dropped.groupby('source')['quantity'].sum()

source
Fulfilment    269.0
Inventory     353.0
Order         632.0
Name: quantity, dtype: float64

The difference between Inventory and Fulfilment  is 84 (353 - 269)

The difference between Order and Fulfilment  is 363 (632 - 269)